libs/frontend/feature-render-builder/src/components/LinkTooltip.vue

```diff
@@ -117,8 +145,30 @@ const updateLink = async () => {
   } else if (componentTargetInput.value === '_blank') {
     removeComponentInput('target')
   }
+}
 
-  await updatePosition()
+const updateProseMirrorLink = () => {
+  // toRaw is necessary because ProseMirror doesn't behave well when proxied
+  // eslint-disable-next-line max-len
+  // https://discuss.prosemirror.net/t/getting-rangeerror-applying-a-mismatched-transaction-even-with-trivial-code/4948/6
+  const { state } = toRaw(editView.value) ?? {}
+  if (state) {
+    const { $cursor } = state.selection as TextSelection
+    if ($cursor) {
+      const cursorPos = $cursor.pos
+      state.doc.nodesBetween(cursorPos, cursorPos, (node, startPos) => {
+        // `nodesBetween` will gives us both root node and link node.
+        // Check `childCount` to make sure it is the link node that gets replaced.
+        if (node.childCount === 0) {
+          const endPos = startPos + node.nodeSize
+          const target = openInNewTab.value ? '_blank' : undefined
+          // TODO: is it necessary & possible to keep other marks? (color, font-weight, font-family, etc)
+          const newLinkNode = createLinkNode(schemaText, editedLink.value, target)
+          editView.value?.dispatch(state.tr.replaceWith(startPos, endPos, newLinkNode))
+        }
+      })
```

Comment:
Hi @sampullman 

The use of `state.doc.nodesBetween()` here is to find & replace the node at the current cursor. It worked, but it seems awkward 😬. I think there should be a better way!

Reply:
I think `nodesBetween` works, but I'm not sure `node.childCount === 0` always works if there are overlapping nodes.  It may better to check which node has a `link` mark present. Another thing you can do to get the node is `$cursor.parent.child($cursor.index())`.

While thinking about this I noticed if there is another mark within the link text area (like text color), only that inner node text will be replaced with the above approach. So changing `https://google.com` to `https://test.com` might result in `https://https://test.com.com`, if the `google` part is styled. My hacky solution is to check nodes before/after the current one to see if they have a link mark, and adjust the start/end replacement position.

For retaining existing marks, I would need to test a few things, but I think it's possible in a limited way. We can add some code to check each node we're replacing, and if the new link has the same substring (at the same start position), skip it instead of replacing. However, I think this is a lot more effort than it's worth right now.

Finally, I've been considering how to do things like "style all links the same way automatically". I think one option is to have some special theme variables that we resolve when links are created, and apply them to the mark. For example `${link-color}`, `${link-visited-color}`, etc. It's a low priority feature, though!